system/security/LdapSecurity/ldapsecurity.ipp

```diff
@@ -370,50 +376,51 @@ public:
     virtual ~CLdapSecManager();
 
 //interface ISecManager : extends IInterface
-    ISecUser * createUser(const char * user_name);
-    ISecResourceList * createResourceList(const char * rlname);
-    bool subscribe(ISecAuthenticEvents & events);
-    bool unsubscribe(ISecAuthenticEvents & events);
-    bool authorize(ISecUser& sec_user, ISecResourceList * Resources, IEspSecureContext* secureContext);
-    bool authorizeEx(SecResourceType rtype, ISecUser& sec_user, ISecResourceList * Resources, IEspSecureContext* secureContext = NULL);
-    SecAccessFlags authorizeEx(SecResourceType rtype, ISecUser& sec_user, const char* resourcename, IEspSecureContext* secureContext = NULL);
-    virtual SecAccessFlags authorizeFileScope(ISecUser & user, const char * filescope);
-    virtual bool authorizeFileScope(ISecUser & user, ISecResourceList * resources);
-    virtual SecAccessFlags authorizeWorkunitScope(ISecUser & user, const char * wuscope);
+    SecFeatureSet queryFeatures(SecFeatureSupportLevel level) const override;
+    ISecUser * createUser(const char * user_name, IEspSecureContext* secureContext = nullptr);
+    ISecResourceList * createResourceList(const char * rlname, IEspSecureContext* secureContext = nullptr);
+    bool subscribe(ISecAuthenticEvents & events, IEspSecureContext* secureContext = nullptr);
+    bool unsubscribe(ISecAuthenticEvents & events, IEspSecureContext* secureContext = nullptr);
+    bool authorize(ISecUser& sec_user, ISecResourceList * Resources, IEspSecureContext* secureContext = nullptr);
+    bool authorizeEx(SecResourceType rtype, ISecUser& sec_user, ISecResourceList * Resources, IEspSecureContext* secureContext = nullptr);
+    SecAccessFlags authorizeEx(SecResourceType rtype, ISecUser& sec_user, const char* resourcename, IEspSecureContext* secureContext = nullptr);
+    virtual SecAccessFlags authorizeFileScope(ISecUser & user, const char * filescope, IEspSecureContext* secureContext = nullptr);
+    virtual bool authorizeFileScope(ISecUser & user, ISecResourceList * resources, IEspSecureContext* secureContext = nullptr);
+    virtual SecAccessFlags authorizeWorkunitScope(ISecUser & user, const char * wuscope, IEspSecureContext* secureContext = nullptr);
```

Comment:
I agree, which is why the methods I added use override instead of virtual. I wasn't sure that making what is essentially a logically transparent change to existing signatures was in the scope of this change, or if it would be viewed like replacing tabs (valid but unnecessary changes that the make it harder to recognize the intent of the PR).